devtools/compat/compat.py

```diff
@@ -0,0 +1,291 @@
+curr_tag_v0 = get_image_tag_by_task_def(v0_def)
+curr_tag_v1 = get_image_tag_by_task_def(v1_def)
+curr_tag_v2 = get_image_tag_by_task_def(v2_def)
+curr_tag_v3 = get_image_tag_by_task_def(v3_def)
+curr_tag_f0 = get_image_tag_by_task_def(f0_def)
+curr_tag_f1 = get_image_tag_by_task_def(f1_def)
+
+print("Network is currently running the following versions:")
+print(f"\t{v0_fam}={curr_tag_v0}")
+print(f"\t{v1_fam}={curr_tag_v1}")
+print(f"\t{v2_fam}={curr_tag_v2}")
+print(f"\t{v3_fam}={curr_tag_v3}")
+print(f"\t{f0_fam}={curr_tag_f0}")
+print(f"\t{f1_fam}={curr_tag_f1}\n")
+
+
+if TEST_MODE == "manual":
+    print("\n------------ Starting MANUAL compat test ------------\n")
+    print("Upgrade path:")
+    print("\tvault (force)\t==> latest")
+    print(f"\tvalidators 0, 1\t==> {ALT_TEST_TAG}")
+    print(f"\tvalidators 2, 3\t==> {TEST_TAG}")
+    print(f"\tfullnodes 0, 1\t==> {ALT_TEST_TAG}\n")
+    new_v0_def = update_task_def_image_tags(v0_def, ALT_TEST_TAG)
+    new_v1_def = update_task_def_image_tags(v1_def, ALT_TEST_TAG)
+    new_f0_def = update_task_def_image_tags(f0_def, ALT_TEST_TAG)
+    new_f1_def = update_task_def_image_tags(f1_def, ALT_TEST_TAG)
+    new_v2_def = update_task_def_image_tags(v2_def, TEST_TAG)
+    new_v3_def = update_task_def_image_tags(v3_def, TEST_TAG)
+
+    if args.dry:
+        print(f"DRY RUN: updated validators/fullnodes 0,1 to {ALT_TEST_TAG}")
+        print(f"DRY RUN: updated validators 2,3 to {TEST_TAG}")
+    else:
+        rev_v0 = register_task_def_update(v0_fam, new_v0_def)
+        rev_v1 = register_task_def_update(v1_fam, new_v1_def)
+        rev_f0 = register_task_def_update(f0_fam, new_f0_def)
+        rev_f1 = register_task_def_update(f1_fam, new_f1_def)
+        rev_v2 = register_task_def_update(v2_fam, new_v2_def)
+        rev_v3 = register_task_def_update(v3_fam, new_v3_def)
+
+        update_service_force(WORKSPACE, vault_fam)
+
+        update_service_task_def(WORKSPACE, v0_fam, rev_v0)
+        update_service_task_def(WORKSPACE, v1_fam, rev_v1)
+        update_service_task_def(WORKSPACE, v2_fam, rev_v2)
+        update_service_task_def(WORKSPACE, v3_fam, rev_v3)
+
+        print("Wait 30 seconds before updating fullnodes.")
+        time.sleep(30)
+
+        update_service_task_def(WORKSPACE, f0_fam, rev_f0)
+        update_service_task_def(WORKSPACE, f1_fam, rev_f1)
+
+elif TEST_MODE == "testnet":
+    print("\n------------ Starting TESTNET compat test ------------\n")
+    print("Upgrade path:")
+    print("\tvault (force)\t==> latest")
+    print(f"\tvalidators 0, 1\t==> {TEST_TAG}")
+    print(f"\tvalidators 2, 3\t==> {ALT_TEST_TAG}")
+    print(f"\tfullnodes 0, 1\t==> {ALT_TEST_TAG}\n")
+    print("Diagnosing node health...")
+    unhealthy = False
+    all_tags = [
+        curr_tag_v0,
+        curr_tag_v1,
+        curr_tag_v2,
+        curr_tag_v3,
+        curr_tag_f0,
+        curr_tag_f1,
+    ]
+    if not all([tag == ALT_TEST_TAG for tag in all_tags]):
+        unhealthy = True
```

Comment:
I think the node health actually refers to if the ECS task is using the intended image tag.  I think we should add other checks, such as containers in the node are actually up. 

It looks like what we have here is setting up the skeleton. I supposed storage, consensus, client checks would be added later to determine backward compatibility breakage?

Reply:
Yea, I should change up the terminology. We already have some stuff for checking container up for devnet right? 

And yea the idea was that initial run of this tool will lock/bring up the necessary network for test. Then other tests (don't exist yet) can do their query/submit txn/ etc to the network to determine breakages. I imagine pylibra would be a good way to test this. Plus reading metrics from prometheus

Reply:
alrighty, land this as is and add more tests later.